src/Containers/FileUpload/FileUpload.tsx

```diff
@@ -0,0 +1,236 @@
+    const [opacityLoading,setOpacityLoading]=useState(0)
+    const [positionIsSuccess,setPositionIsSuccess]=useState(!isSuccess)
+    const [opacityIsSuccess,setOpacityIsSuccess]=useState(0)
+    const [loadingContainerHeight,setLoadingContainerHeight]=useState(0)
+    const [widthComponent,setWidthComponent]=useState(0)
+    const [margin]=useState(10)
+    const [widthIsSuccess,setWidthIsSuccess]=useState(0)
+    const [positionIsFailure,setPositionIsFailure]=useState(!isFailure)
+    const [opacityIsFailure,setOpacityIsFailure]=useState(0)
+
+    // this sets height of the component, is used to transition between heights. 
+    useEffect(()=>{
+        if(isUploading||isSuccess||isFailure){
+            if(totalHeightPlus){
+                setHeight(totalHeightPlus)
+            }else{
+                setHeight(undefined)
+                setMaxHeight(600)
+            }
+        }else if(totalHeight){
+            setHeight(totalHeight)   
+        }
+    },[isUploading,isSuccess,isFailure,totalHeight,totalHeightPlus])
+
+    // this is to calculate (set) some values
+    useEffect(()=>{
+        if(containerRef.current?.scrollHeight){
+            const contentHeight=containerRef.current.scrollHeight
+            setTotalHeight(contentHeight-padding*2)
+            setMaxHeight(contentHeight-padding*2)
+        }
+        if(rootRef.current?.clientWidth){
+            setWidthComponent(rootRef.current?.clientWidth-margin*2-padding*2)
+        }
+    },[])
+
+    // this is to set some values the first time when the component it's expanded
+    useEffect(()=>{
+        if(height===undefined&&(isUploading||isSuccess||isFailure)){ 
+            if(containerRef.current?.scrollHeight){
+                setTotalHeightPlus(containerRef.current.scrollHeight-padding*2)
+            }
+            if(loadingContainerRef.current?.getBoundingClientRect().top){
+                setPositionTopLoading(loadingContainerRef.current?.getBoundingClientRect().top-margin)
+            }
+        }
+    },[height,isUploading,isSuccess,isFailure])
+
+    // this is to calculate and set the width of the success and failure container component
+    useEffect(()=>{
+        if(isSuccess||isFailure){
+            const width=containerRef.current?.getBoundingClientRect().width
+            if(width){
+                setWidthIsSuccess(width-padding*2-margin*2)
+            }
+        }
+    },[isSuccess,isFailure])
+
+    // this is to fade out uploading container component
+    useEffect(()=>{
+        if(!isUploading||isSuccess||isFailure){
+            setOpacityLoading(0)
+            setPositionLoading(true)
+        }
+        return ()=>{
+            setPositionLoading(false)
+            setOpacityLoading(1)
+        }
+    },[isUploading,isSuccess,isFailure])
+
+    // this is to fade out success container component
+    useEffect(()=>{
+        if(!isSuccess||isFailure){
+            setOpacityIsSuccess(0)
```

Comment:
Why is opacity have loading / success states? Can the naming be improved

Reply:
because there is opacity for the isUploading panel, opacity for the isSuccess panel and opacity for the isFailure panel